### PR TITLE
Avoided unnecessary calls to string translation methods for catalog/sales rules labels

### PR DESCRIPTION
**Pull request by luigifab**
### Description

This PR stop translations of rules labels from frontend.
Perhaps the best way is to remove all translations from model, and translate only from blocks? I don't know.

Here a before/after with Blackfire with a cart with ~50 products (simple, configurable, bundles):

original (before):
big-cart-before

after without the static property:
big-cart-after-effect

after:
static
big-cart-after

Can fix partially #3127 or not.
Tested with PHP 8.0 and 8.2 with OM rc2.

If you have others modules or custom dev, that do the same, you need to update your files to have better performance on your website. Example with [HiPay module's](https://github.com/hipay/hipay-fullservice-sdk-magento1/blob/master/src/app/code/community/Allopass/Hipay/Model/Rule/Condition/Product/Subselect.php).

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by fballiano** [comment]
wow, I wasn't expecting the `__` to generate all those calls, I thought it would immediately hit a cache :\

**Review by fballiano — APPROVED** [review]

**Comment by joshua-bn** [comment]
Does it make any difference for time?

**Comment by fballiano** [comment]
if I read the screenshots correctly it goes from 5secs to 3.6secs (with @luigifab's DB data obviously)

File: app/code/core/Mage/CatalogRule/Model/Rule/Action/Product.php [hunk 1]
```diff
@@ -42,10 +42,10 @@ public function loadAttributeOptions()
     public function loadOperatorOptions()
     {
         $this->setOperatorOption([
-            'to_fixed' => Mage::helper('cataloginventory')->__('To Fixed Value'),
-            'to_percent' => Mage::helper('cataloginventory')->__('To Percentage'),
-            'by_fixed' => Mage::helper('cataloginventory')->__('By Fixed value'),
-            'by_percent' => Mage::helper('cataloginventory')->__('By Percentage'),
+            'to_fixed'   => static::$translate ? Mage::helper('cataloginventory')->__('To Fixed Value') : 'To Fixed Value',
```
Comment (kiatng) [inline app/code/core/Mage/CatalogRule/Model/Rule/Action/Product.php:45]: What is `static::$translate`? I cannot find its reference, also:

> Undefined property '$translate'.intelephense(1014)

File: app/code/core/Mage/CatalogRule/Model/Rule/Action/Product.php [hunk 2]
```diff
@@ -42,10 +42,10 @@ public function loadAttributeOptions()
...
+            'to_percent' => static::$translate ? Mage::helper('cataloginventory')->__('To Percentage') : 'To Percentage',
+            'by_fixed'   => static::$translate ? Mage::helper('cataloginventory')->__('By Fixed value') : 'By Fixed value',
+            'by_percent' => static::$translate ? Mage::helper('cataloginventory')->__('By Percentage') : 'By Percentage',
         ]);
```
Comment (kiatng) [inline app/code/core/Mage/CatalogRule/Model/Rule/Action/Product.php:49]: Not sure if this works:
```suggestion
        if (static::$isAdmin) {
            $this->setOperatorOption([
                'to_fixed' => Mage::helper('cataloginventory')->__('To Fixed Value'),
                'to_percent' => Mage::helper('cataloginventory')->__('To Percentage'),
                'by_fixed' => Mage::helper('cataloginventory')->__('By Fixed value'),
                'by_percent' => Mage::helper('cataloginventory')->__('By Percentage'),
            ]);
        } else {
            $this->setOperatorOption([]);
       }
```
`$isAdmin` is more meaningful as `$translate` is applicable in frontend.  It reduces the number of comparisons to just one.

File: app/code/core/Mage/Rule/Model/Action/Abstract.php [hunk 3]
```diff
@@ -36,8 +36,20 @@
  */
 abstract class Mage_Rule_Model_Action_Abstract extends Varien_Object implements Mage_Rule_Model_Action_Interface
 {
+    /**
+     * Flag to enable translation for loadOperatorOptions/loadValueOptions/loadAggregatorOptions/getDefaultOperatorOptions
+     * It's useless to translate these data on frontend
+     *
+     * @var bool
+     */
+    protected static $translate;
```
Comment (kiatng) [inline app/code/core/Mage/Rule/Model/Action/Abstract.php:45]:
```suggestion
    protected static $isAdmin;
```

**Comment by kiatng** [comment]
I think it's better to stop the translation in here:
https://github.com/OpenMage/magento-lts/blob/6851e17f5f74f82241762b8f1c0779e6f0d54b51/app/code/core/Mage/Rule/Model/Action/Abstract.php#L39-L52

To something like:
```php
    public function __construct()
    {
        parent::__construct();
        if (!Mage::app()->getStore()->isAdmin()) {
            return;
        }
        $this->loadAttributeOptions()->loadOperatorOptions()->loadValueOptions();

        // ...
    }
```

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**